### PR TITLE
update NativeTokenConstraint; add Polygon whales

### PR DESCRIPTION
**Pull request by scott-silver**
Updates to NativeTokenConstraint to work with Polygon deploy

### CODE REVIEW

File: scenario/constraints/NativeTokenConstraint.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export class NativeTokenConstraint<T extends CometContext, R extends Requirement
       async function (ctx: T): Promise<T> {
         for (const symbol in ctx.assets) {
           const contract = await ctx.world.deploymentManager.contract(symbol);
-          if (contract.deposit) {
+          if (contract['deposit()']) {
```
Comment (scott-silver) [inline scenario/constraints/NativeTokenConstraint.ts:12]: Polygon USDC has a `deposit` function, but it takes multiple arguments:

https://polygonscan.com/address/0xdd9185db084f5c4fff3b4f70e7ba62123b812226#code

Updating this to check for the deposit with the function signature we expect

File: src/deploy/index.ts [hunk 2]
```diff
@@ -88,6 +88,11 @@ export const WHALES = {
     '0x2775b1c75658be0f640272ccb8c72ac986009e38',
     '0x1a9c8182c09f50c8318d769245bea52c32be35bc',
     '0x3d9819210a31b4961b30ef54be2aed79b9c9cd3b',
+  ],
+  polygon: [
+    '0x2093b4281990a568c9d588b8bce3bfd7a1557ebd', // WETH whale
```
Comment (scott-silver) [inline src/deploy/index.ts:93]: The NativeTokenConstraint needs there to be at least one whale so that it can transfer funds to it

**Review by kevincheng96 — APPROVED** [review]

**Comment by kevincheng96** [comment]
Are the mumbai scenario errors in CI related to these changes?

File: scenario/constraints/NativeTokenConstraint.ts (hunk 1 above)
Comment (jflatow) [reply scenario/constraints/NativeTokenConstraint.ts:12]: 🧼

**Comment by scott-silver** [comment]
@kevincheng96 I'm not sure, I haven't looked into the Mumbai errors

**Review by jflatow — APPROVED** [review]

**Pull request merged**